### DISCUSSION TITLE
issue when run code in RDP or VPS while bypassing cloudflare (can't bypass capcha that appear after click on button) multiple iframe

### DISCUSSION THREAD

**Discussion by Akshay-Shingala** [seleniumbase/SeleniumBase#3344] [General]
There is a website where I have to bypass Cloudflare two times: the first time when I load the website and the second time when I click on some button, then also one Cloudflare CAPTCHA opens, but I am not able to bypass both if I am using uc_gui_click_captcha to solve the first captcha I am not able to solve it a second time.

```from seleniumbase import SB
import time
from selenium.webdriver.common.by import By
from selenium.webdriver.support.ui import WebDriverWait
from selenium.webdriver.support import expected_conditions as EC

fire = '/html/body/div[1]/div/main/div/div/div[1]/div/div[1]/div[4]/div/div[3]/button[4]'
# with SB(uc=True, test=True, ad_block=True) as sb:

with SB(uc=True, test=True,proxy="",guest=True,uc_sub=True,uc_subprocess=True,) as sb:
    url = ""

    sb.activate_cdp_mode(url)
    sb.uc_gui_click_captcha()
    time.sleep(5)
    sb.uc_click("button.custom-pr2mrc", reconnect_time=15,timeout=15)
    sb.uc_gui_handle_captcha(frame="iframe")
    sb.uc_gui_click_captcha()
    
    sb.uc_click("button.custom-pr2mrc", reconnect_time=15)
    sb.uc_gui_handle_cf(frame="iframe")
 ```
In VPS, I am using a proxy. using sb.uc_gui_click_captcha(), it can bypass first page but when it try to click on button

![image](https://github.com/user-attachments/assets/c21fd081-0701-4451-a725-ebadb3afa0b5)

This Cloudflare checkbox appears, and I am not able to bypass it. If I run code locally (windows 11), it can bypass both checkboxes, but in VPS (windows 11), it creates an issue. Also, on the page, there are more than one iframe. Did this create an issue for solving the checkbox?

I want to run multiple windows simultaneously.

can any one help me?

**Comment by mdmintz** [comment]
See https://github.com/seleniumbase/SeleniumBase/issues/3213#issuecomment-2433487801

**Reply by Akshay-Shingala** [reply]
I can't bypass Cloudflare on my Windows VPS, but I'm keeping the session open while using the script. 
Is there any way to bypass Cloudflare on a VPS? Could I use a different operating system, or do you have any other suggestions?

**Comment by krzysztofkub** [comment]
I was able to run pyautogui in VNC by using Xvfb virtual display and assigning it to pyautogui:

pyautogui._pyautogui_x11._display = Xlib.display.Display(display)

    with SB(
            uc=True,
            incognito=True,
            locale_code="en",
            headed=True,
            xvfb=True,
    ) as sb:
    
    Needed to click second captcha with pyautogui.click

**Reply by mdmintz** [reply]
Still going to hit https://github.com/asweigart/pyautogui/issues/284. Run directly from a server, such as with GitHub Actions.

**Reply by IndrajeethY** [reply]
I tried to run this on a server but it cant pass through captcha still, its a linux vps,

**Comment by Akshay-Shingala** [comment]
I can't bypass Cloudflare on my Windows VPS, but I'm keeping the session open while using the script.
Is there any way to bypass Cloudflare on a VPS? Could I use a different operating system, or do you have any other suggestions?
